vision_agent_tools/models/model_registry.py

```diff
@@ -1,19 +1,63 @@
-from typing import Dict, Callable
+from typing import List, Dict, Type, Callable
+from enum import Enum
 from vision_agent_tools.shared_types import BaseMLModel
 from vision_agent_tools.models.florencev2 import Florencev2
+from vision_agent_tools.models.owlv2 import Owlv2
 
 
-MODEL_REGISTRY: Dict[str, Callable[[], BaseMLModel]] = {"florence2": Florencev2}
+class TextToObjectDetectionModel(str, Enum):
+    FLORENCE_V2 = "florencev2"
+    OWLV2 = "owlv2"
 
 
-def get_model(model_name: str) -> BaseMLModel:
+MODEL_REGISTRY: Dict[str, Callable[[], BaseMLModel]] = {
+    TextToObjectDetectionModel.FLORENCE_V2.value: Florencev2,
+    TextToObjectDetectionModel.OWLV2.value: Owlv2,
+}
+
+TASK_MODEL_MAP: Dict[Enum, List[str]] = {
+    TextToObjectDetectionModel: [model.value for model in TextToObjectDetectionModel]
+}
+
+
+def get_model(model_name: str, task: Type[Enum]) -> BaseMLModel:
     """
-    Retrieve a model from the registry based on the model name
+    Retrieve a model from the registry based on the model name and task
+
+    Args:
+        model_name (str): The name of the model to retrieve
+        task (Type[Enum]): The enum representing the valid models for a specific task
+
+    Returns:
+        BaseMLModel: An instance of the requested model
+
+    Raises:
+        ValueError: If the model is not registered or is not valid for the given task
     """
 
+    if model_name not in TASK_MODEL_MAP[task]:
+        raise ValueError(
+            f"Model '{model_name}' is not valid for the task '{task.__name__}'."
+        )
+
     model_class = MODEL_REGISTRY.get(model_name)
 
     if not model_class:
-        raise ValueError(f"Model '{model_name}' is not registered")
+        raise ValueError(
+            f"Model '{model_name}' is not registered in the model registry."
+        )
 
     return model_class()
```

Comment:
One question, Is the model being initialized here? If that's the case, I think we should fix this to only return the model.

Reply:
makes sense to change it to first bring the class then initialize. I am updating the code:

`
def get_model_class(model_name: str, task: Type[Enum]) -> BaseMLModel:
...
return model_class`

`def __init__(self, model: str):
        self.model_name = model
        model_class = get_model_class(model_name=model, task=TextToObjectDetectionModel)
        model_instance = model_class()
        super().__init__(model=model_instance)`